src/__tests__/components/fields/text-field/text-field.spec.tsx

```diff
@@ -105,9 +107,60 @@ describe(DEFAULT_FIELD_TYPE, () => {
 		expect(getTextfield()).toHaveAttribute("readonly");
 		expect(getTextfield()).toBeDisabled();
 	});
+
+	it("should not prevent copy and paste", async () => {
+		renderComponent({
+			customOptions: {
+				preventCopyAndPaste: false,
+			},
+		});
+		const textField = getTextfield();
+		textField.focus();
+		await act(async () => {
+			await userEvent.paste(EXPECTED_TEXT);
+			expect(textField).toHaveValue(EXPECTED_TEXT);
+		});
```

Comment:
I don't see the need for `act()` since jest is not complaining of any async updates running in the background, you can try this instead:

```
await waitFor(() => userEvent.paste(EXPECTED_TEXT));
expect(textField).toHaveValue(EXPECTED_TEXT);
```